src/core/plugin/luapi_application.h

```diff
@@ -1976,6 +1978,193 @@ static int applib_openFile(lua_State* L) {
+        }
+
+        xoj::util::GObjectSPtr<GFile> file(g_file_new_for_path(path), xoj::util::adopt);
+        if (!g_file_query_exists(file.get(), NULL)) {
+            lua_pop(L, 8);  // pop the params we fetched from the global param-table from the stack
+            lua_pushfstring(L, "Error: file '%s' does not exist.", path);  // soft error
+            continue;
+        }
+
+        XojPageView* pv = control->getWindow()->getXournal()->getViewFor(control->getCurrentPageNo());
+        ImageHandler imgHandler(control, pv);
+        auto [img, width, height] = imgHandler.createImage(file.get(), x, y);
+        if (!img) {
+            lua_pop(L, 8);  // pop the params we fetched from the global param-table from the stack
+            lua_pushfstring(L, "Error: creating the image (%s) failed.", path);  // soft error
+            continue;
+        }
+
+        // apply width/height parameter
+        if (maxWidthParam != -1 && maxHeightParam != -1) {
+            width = maxWidthParam;
+            height = maxHeightParam;
+        } else if (maxWidthParam != -1 && maxHeightParam == -1) {
+            if (aspectRatio) {
+                maxHeightParam = static_cast<int>(
+                        std::round(static_cast<double>(height) / static_cast<double>(width) * maxWidthParam));
+                height = maxHeightParam;
+            }
+            width = maxWidthParam;
+        } else if (maxHeightParam != -1 && maxWidthParam == -1) {
+            if (aspectRatio) {
+                maxWidthParam = static_cast<int>(
+                        std::round(static_cast<double>(width) / static_cast<double>(height) * maxHeightParam));
+                width = maxWidthParam;
+            }
+            height = maxHeightParam;
+        }
+
+        // apply scale option
+        width = static_cast<int>(std::round(width * scale));
+        height = static_cast<int>(std::round(height * scale));
+
+        // scale down keeping the current aspect ratio after the manual scaling to fit the image on the page
+        // if the image already fits on the screen, no other scaling is applied here
+        // already sets width/height in the image
+        imgHandler.automaticScaling(img, x, y, width, height);
+
+        // store the image to later build the undo/redo action chain
+        images.push_back(img);
+
+        bool succ = imgHandler.addImageToDocument(img, false);
+        if (!succ) {
+            lua_pop(L, 8);  // pop the params we fetched from the global param-table from the stack
+            lua_pushfstring(L, "Error: Inserting the image (%s) failed.", path);  // soft error
+            continue;
+        }
+
+        lua_pop(L, 8);  // pop the params we fetched from the global param-table from the stack
+        lua_pushnil(L);
+    }
+
+    // Check how the user wants to handle undoing
+    lua_getfield(L, 1, "allowUndoRedoAction");
+    const char* allowUndoRedoAction = luaL_optstring(L, -1, "grouped");
```

Comment:
You need to pop this value again. Otherwise it will be on the stack when returning.

```suggestion
    lua_getfield(L, 1, "allowUndoRedoAction");
    const char* allowUndoRedoAction = luaL_optstring(L, -1, "grouped");
    lua_pop(L, 1);
```

Reply:
Yes you're right. Thanks for noting (cannot believe I didn't note this when testing the function)